templates/logrotate.j2

```diff
@@ -0,0 +1,16 @@
+{{logs_directory}}/*.log
+{
+  rotate 10
+  size 100M
+  missingok
+  notifempty
+  create 0600 {{mongo_user}} {{mongo_user}}
+  compress
+  nomail
+  nocopytruncate
+  sharedscripts
+  postrotate
+    sudo /bin/kill -SIGUSR1 $(sudo systemctl show -p MainPID --value snap.charmed-mongodb.mongod.service)
```

Comment:
From what I understand about Unix / Linux log rotate it works when the kill -SIGUSR1 signal is sent to the process that writes logs(in this case mongod) and then the logs are rotated according to what settings are specified in `/etc/logrotate.d/mongodb`. 

Is this correct?

If so I have concerns about us killing `mongod` and in some cases forcing a primary re-election. I understand that this is similar to what [percona does](https://stackoverflow.com/questions/20162176/centos-linux-setting-logrotate-to-maximum-file-size-for-all-logs)

But is that what [mysql does](https://github.com/canonical/mysql-operator/blob/14e7dd5662a8b4fe224b59bc453fb3bda4738466/templates/logrotate.j2)? I genuinely can't tell what their script does (without doing a deep dive) 

Either way, I think we can do better. If we have this cron job running anyway, why not just delete the log file manually and then let mongodb write a new one, rather than sending SIGKILL? This would give us better performance regarding primary elections.

I could be naive here and missing a bigger picture though. But I am curious what your thoughts are @delgod and @Mehdi-Bendriss

Reply:
This command does  not kill mongodb process, it only sends singal to it to rotate the logs, that's it

Reply:
Mongodb will rotate logs according to [these](https://github.com/canonical/mongodb-operator/pull/409/files#diff-094641a8acb9699cf7cfb0984347bf0e0abf254adf46b8080311fc973a03daffR142) settings,

and `/etc/logrotate.d/mongodb` is configuration for `logrotate` only and not for mongo

Reply:
Also : 
When reporting activity data to a log file, by default, MongoDB only rotates logs in response to the [logRotate](https://www.mongodb.com/docs/manual/reference/command/logRotate/#mongodb-dbcommand-dbcmd.logRotate) command, or when the [mongod](https://www.mongodb.com/docs/manual/reference/program/mongod/#mongodb-binary-bin.mongod) or [mongos](https://www.mongodb.com/docs/manual/reference/program/mongos/#mongodb-binary-bin.mongos) process receives a SIGUSR1 signal from the operating system. Both server logs and [audit logs](https://www.mongodb.com/docs/manual/core/auditing/#std-label-auditing) may be rotated with the [logRotate](https://www.mongodb.com/docs/manual/reference/command/logRotate/#mongodb-dbcommand-dbcmd.logRotate) command, either together or independently.